### DISCUSSION TITLE
Unable to get cli working

### DISCUSSION THREAD

**Discussion by Sharaddition** [ente-io/ente#3883] [Q&A] [answered]
After I set env variable, I'm getting following error:
```sh
PS C:\Users\dell\Documents\ente> .\ente
2024/10/29 15:18:03 Failed to open db C:\Users\dell\Documents\ente\config.yaml/ente-cli.db open C:\Users\dell\Documents\ente\config.yaml/ente-cli.db: The system cannot find the path specified.
PS C:\Users\dell\Documents\ente> ls


    Directory: C:\Users\dell\Documents\ente


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----        10/29/2024   3:01 PM                export
-a----        10/29/2024   2:57 PM             48 config.yaml
-a----        10/23/2024   3:43 AM        9546240 ente.exe
```

**Comment by mngshm** [comment]
Hey, can you share what command you used to set the env variable ? The problem is quiet unclear from what you've mentioned. 

It can't open `C:\Users\dell\Documents\ente\config.yaml/ente-cli.db ` because `config.yaml` is itself a file already. There can't be another file after `config.yaml` in the absolute path. Please check your path once again.

**Reply by Sharaddition** [reply]
Hey @mngshm, I'm using powershell and this is my command:
```sh
$env:ENTE_CLI_CONFIG_PATH = "C:\Users\dell\Documents\ente\config.yaml"
```

later I tried following command, still same problem:
```sh
setx ENTE_CLI_CONFIG_PATH=C:\Users\dell\Documents\ente\config.yaml
```
Is there any problem in this command?

**Reply by mngshm** [reply]
Did you restart powershell ? The Shell might need a hard restart (or a reboot of the machine, idk as I don't have any experience with powershell) so that the previous env variable value is reset to nil. 

The first command, looks good now without the `config.yaml/ente-cli=.db` (which you had entered previously like you mentioned in the first message on this discussion).

Edit: I have been able to reproduce this on my system. Will get back to you soon as I get a fix.

**Reply by mngshm** [reply, marked as answer]
Hey, there's a bit of confusion in the documentation. I feel especially in the `ente/cli/config.yaml.example`. 

So, here's what I found. The `~/.ente` dir in Linux contains the ente-cli.db database file. There must be something similar created in the windows side as well. Please find that directory and then set the env variable again to that directory and not to the `config.yaml` inside it. This should be able to fix it, as it did on mine. I am attaching a secret of my terminal. 

I was able to run the ente command after that which is added to my $PATH in the system. 

![Capture-Oct-29-18-45-41](https://github.com/user-attachments/assets/5be1f914-8190-4d4d-b8d3-db73a0b3b0b5)

Let me know if you need more help.

**Reply by Sharaddition** [reply]
Thanks @mngshm for pointing out the problem. I was setting `config.yaml` as env. var. After fixing that now everything is working perfectly.

**Comment by mngshm** [comment]
Just an update, in one of the [recent commits](https://github.com/ente-io/ente/commit/82c1987a6ddd04b1c943c47021081c3942d7bec2) last week we changed the variable name `ENTE_CLI_CONFIG_PATH` to `ENTE_CLI_CONFIG_DIR`. Although, it won't be an issue as it is backward compatible.

Closing this discussion.
